_posts/2021-06-06-spring-camel.md

```diff
@@ -0,0 +1,422 @@
+
+These are loaded in the Camel context and are used to execute the routing logic when the route is triggered. Each route is identified by a unique identifier in the Camel context.
+
+**Endpoints represent the source and destination of a message.** They are usually referred to in the Domain Specific Language (DSL) via their URIs. Examples of endpoint is either a URI or URL in a web application or a Destination in a JMS system.
+
+### Domain Specific Language (DSL)
+We define routes in Apache Camel with a variety of [Domain Specific Languages (DSL)](https://camel.apache.org/manual/latest/dsl.html). Java DSL, and Spring XML DSL are the two main types of DSLs used in Spring applications.  
+
+Here is an example of a route defined using a Java DSL :
+
+```java
+("file:/mysrc").split().tokenize("\n").to("jms:queue:myQueue")
+```
+Here we have defined a route with a file endpoint as a source and a JMS queue as a destination. When we execute this route, it will read all files from the input folder `mysrc`, pick up each file due to the `split` method, and call `tokenize` method on the contents of each file with a newline separator, and finally send each line to the JMS queue.
+
+The same route defined using Spring XML DSL looks like this :
+```xml
+
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="
+       http://www.springframework.org/schema/beans 
+       http://www.springframework.org/schema/beans/spring-beans.xsd
+       http://camel.apache.org/schema/spring 
+       http://camel.apache.org/schema/spring/camel-spring.xsd
+    ">
+
+  <camelContext id="sendtoqueue" xmlns="http://camel.apache.org/schema/spring">
+    <route>
+      <from uri="file:/myFolder"/>
+      <split/>
+      <tokenize token="\n"/>
+      <to uri="jms:queue:myQueue"/>
+    </route>
+  </camelContext>
+
+</beans>
+```
+
+### Components
+
+The transport of a message from the source to the destination goes through multiple steps. Processing in each step might require accessing different types of resources in the message flow like invocation of a bean method or calling an API. **We use components to perform the function of connecting to these resources.** 
+
+
+For example, the route defined with the below Java DSL uses the `file` component to bridge to the file system and `jms` component to bridge to the JMS provider. 
+
+```java
+("file:/mysrc").split().tokenize("\n").to("jms:queue:myQueue")
+```
+
+Camel has several [pre-built components](https://camel.apache.org/components/latest/) and many others built by communities. Here is a snippet of the components available in Camel which gives us an idea of the wide range of systems we can integrate using the framework:
+
+ | | | | | |
+-|-|-|-|-|-|-
+ActiveMQ|AMQP|Async HTTP Client|Atom|Avro RPC|AWS2 DynamoDB|AWS2 Lambda
```

Comment:
Can you convert this into a simple unordered list? Even if it's long it looks cleaner than this table.